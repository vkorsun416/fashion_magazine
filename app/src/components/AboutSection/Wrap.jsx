import React from 'react';
import css from "./Wrap.module.css";

function Wrap(props) {
    const {children} = props;

    return (
        <div className={css.wrapper}>
            <div className={css.container}>
                <div className={css.grid}>
                    {children}
                </div>
            </div>
        </div>
    );
}

export default Wrap;
