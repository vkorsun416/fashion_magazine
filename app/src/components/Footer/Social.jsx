import React from 'react';
import css from "./Social.module.css";
import {DARK} from "../../layout/themes";

function Social(props) {
    const {theme} = props;

    return (
        <div className={css.wrap}>
            <div className={css.item + " " + (theme === DARK ? css.light : "")}>
                {props.text}
            </div>

            <div className={css.item}>
                <a
                    className={css.link + " " + (theme === DARK ? css.white : "")}
                    href={props.instagram}
                >
                    Instagram
                </a>

                <a
                    className={css.link + " " + (theme === DARK ? css.white : "")}
                    href={props.linkedin}
                >
                    Linkedin
                </a>
            </div>
        </div>
    );
}

export default Social;