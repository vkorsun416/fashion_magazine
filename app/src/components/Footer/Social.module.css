.wrap {
    color: #000;
    display: flex;
    justify-content: space-between;
}

.item {
    padding: 0 50px;
    display: flex;
    justify-content: space-between;
}

.link {
    margin-right: 15px;
    border-bottom: 1px solid #000;
    text-decoration: none;
    color: #000;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
}

.link:last-child {
    padding: 0;
}

.light {
    color: #fff;
}

.white {
    color: white;
    border-bottom: 1px solid #fff;
}

@media screen and (max-width: 850px) {
    .item {
        padding: 0 15px;
    }
}

@media screen and (max-width: 700px) {
    .wrap {
        margin: 0 auto 15px auto;
    }
}