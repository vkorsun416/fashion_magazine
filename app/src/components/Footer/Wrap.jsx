import React, { Component } from "react";
import css from "./Wrap.module.css";
import { DARK } from "../../layout/themes";

class Wrap extends Component {
  renderWrap() {
    const { children, theme, copyright } = this.props;

    return (
      <div className={theme === DARK ? css.black : ""}>
        <div className={css.container}>
          <div className={css.item + " " + (theme === DARK ? css.white : "")}>
            {copyright}
          </div>
          {children}
        </div>
      </div>
    );
  }

  render() {
    return this.renderWrap();
  }
}

export default Wrap;
