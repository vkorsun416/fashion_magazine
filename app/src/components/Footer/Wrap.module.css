.container {
    max-width: 1440px;
    margin: 0 auto;
    padding: 24px;
    display: flex;
    justify-content: space-between;
}

.black {
    background-color: black;
}

.item {
    color: rgba(1, 1, 1, 0.37);
}

.white {
    color: #FFFFFF;
    opacity: 37%;
}

@media screen and (max-width: 700px) {
    .container {
        flex-direction: column-reverse;
    }

    .item {
        margin: 0 auto;
    }
}
