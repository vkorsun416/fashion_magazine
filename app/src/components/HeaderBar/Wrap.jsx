import React, {Component} from 'react';
import css from "./Wrap.module.css";

class Wrap extends Component {
    state = {
        isScrolled: false,
    }

    componentDidMount() {
        window.addEventListener('scroll', () => {
            const isScrolled = window.pageYOffset > document.documentElement.clientHeight;

            if (isScrolled !== this.state.isScrolled) {
                this.setState({isScrolled})
            }
        });
    }

    /**
     *
     * Возвращает фон для хедера,
     * если скролл выше высоты экрана добавляем фон
     */
    getClasses() {
        const {isScrolled} = this.state;
        const classes = [css.rgba];

        if (isScrolled) {
            classes.push(css.blur);
        }

        return classes;
    }

    renderWrap() {
        const {children} = this.props;

        return (
            <header>
                <div className={this.getClasses().join(" ")}>
                    <div className={css.container}>
                        <div className={css.header}>
                            {children}
                        </div>
                    </div>
                </div>
            </header>
        )
    }

    render() {
        return (
            this.renderWrap()
        );
    }
}

export default Wrap;
