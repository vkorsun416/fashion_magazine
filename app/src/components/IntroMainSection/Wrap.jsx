import React from 'react';
import css from "./Wrap.module.css";

function Wrap(props) {
    const {children} = props;

    return (
        <div className={css.wrapper}>
            <div className={css.container}>
                <div className={css.gridWrap}>
                    {children}
                </div>
            </div>
        </div>
    );
}

export default Wrap;
