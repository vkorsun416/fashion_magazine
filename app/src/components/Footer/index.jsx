import React from "react";
import PropTypes from "prop-types";
import Social from "./Social";
import Wrap from "./Wrap";
import * as themes from "../../layout/themes";

function Footer(props) {
    const {theme, className} = props;

    return (
        <div className={className}>
            <Wrap
                theme={theme}
                copyright="Â© 2021 - INTERVIEW / All Rights Reserved"
            >
                <Social
                    text="Find us online"
                    instagram="https://www.instagram.com/skalar.ua/"
                    linkedin="https://www.linkedin.com/company/skalar-it/"
                    theme={theme}
                >
                </Social>
            </Wrap>
        </div>
    );
}

Footer.propTypes = {
    theme: PropTypes.oneOf(Object.values(themes))
}

export default Footer;