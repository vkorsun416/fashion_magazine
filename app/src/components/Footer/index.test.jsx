import React from "react";
import {mount} from "enzyme";
import {StaticRouter} from "react-router-dom";
import Wrap from "./Wrap";
import Social from "./Social";
import {DARK} from "../../layout/themes";
import Footer from ".";

globalThis.console.error = jest.fn();

describe("Footer", () => {
    test("Should render to string", () => {
        const wrap = mount(
            <StaticRouter>
                <Footer />
            </StaticRouter>
        );

        expect(wrap.html()).toMatchSnapshot();
    });

    test("Should render with dependencies", () => {
        const wrap = mount(
            <StaticRouter>
                <Footer />
            </StaticRouter>
        );

        console.log(wrap.find(Wrap).html())

        expect(wrap.find(Wrap)).toHaveLength(1);
        expect(wrap.find(Social)).toHaveLength(1);
        expect(wrap.find(Social).props().theme).toBe(undefined);
    });

    test("Should render with theme", () => {
        const wrap = mount(
            <StaticRouter>
                <Footer theme={DARK}/>
            </StaticRouter>
        );

        expect(wrap.find(Wrap).props().theme).toBe(DARK);
        expect(wrap.find(Social).props().theme).toBe(DARK);
    });

    test("Should not render with wrong theme", () => {
        mount(
            <StaticRouter>
                <Footer theme={"WRONG"}/>
            </StaticRouter>
        );

        expect(globalThis.console.error).toBeCalledTimes(1);
    });

});

