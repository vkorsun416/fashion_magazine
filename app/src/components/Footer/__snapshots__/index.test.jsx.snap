// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Should render to string 1`] = `"<div class=\\"\\"><div class=\\"container\\"><div class=\\"item \\">Â© 2021 - INTERVIEW / All Rights Reserved</div><div class=\\"wrap\\"><div class=\\"item \\">Find us online</div><div class=\\"item\\"><a class=\\"link \\" href=\\"https://www.instagram.com/skalar.ua/\\">Instagram</a><a class=\\"link \\" href=\\"https://www.linkedin.com/company/skalar-it/\\">Linkedin</a></div></div></div></div>"`;
